### ISSUE TITLE
Train stable diffusion

### ISSUE THREAD

**Issue by qibin0506** [keras-team/keras-cv#1773]
How to train my own data through stable diffusion at https://github.com/keras-team/keras-cv/blob/master/keras_cv/models/stable_diffusion?

**Comment by jbischof** [comment]
Yes we have a whole guide for this @qibin0506!

https://keras.io/examples/generative/finetune_stable_diffusion/

**Comment by innat-asj** [comment]
> It's highly recommended that you use a GPU with at least 30GB of memory to execute the code.

It's challenging to train SD with consumer-level GPUs with 16GB VRAM. In Diffusers, it supports gradient-checkpointing, 8-bit optimizer, gradient-accumulation, deep-speed, etc, type of techniques which makes it super easy to train these models.

**Comment by qibin0506** [comment]
\\

> Yes we have a whole guide for this @qibin0506!
> 
> https://keras.io/examples/generative/finetune_stable_diffusion/

Thank you for your reply @jbischof ,but how to train the imageDecoder model with latent?

**Comment by jbischof** [comment]
I'm afraid I'm not super familiar with this workflow. @ianstenbit any insights?

**Comment by innat-asj** [comment]
@qibin0506 cc. @jbischof 
It would be better to rephrase your main query. The decoding stage ([Decoder](https://github.com/keras-team/keras-cv/blob/fc81ab52fca28ccfa5f1ec83523fbe2a4a177be3/keras_cv/models/stable_diffusion/decoder.py#L28)) is only applicable in the inference steps, while [generating image](https://github.com/keras-team/keras-cv/blob/fc81ab52fca28ccfa5f1ec83523fbe2a4a177be3/keras_cv/models/stable_diffusion/stable_diffusion.py#L235-L238) from text. I think under normal circumstances, it is not part of the training process. While following [this](https://keras.io/examples/generative/finetune_stable_diffusion/) code example with your data (image-prompt), which components you think are absent for training or fine-tuning stable diffusion workflow?


![2uC8rYJ](https://github.com/keras-team/keras-cv/assets/81627860/6d843ba6-e994-421e-9753-90a6ed76c00a)

**Comment by qibin0506** [comment]
> @qibin0506 cc. @jbischof It would be better to rephrase your main query. The decoding stage ([Decoder](https://github.com/keras-team/keras-cv/blob/fc81ab52fca28ccfa5f1ec83523fbe2a4a177be3/keras_cv/models/stable_diffusion/decoder.py#L28)) is only applicable in the inference steps, while [generating image](https://github.com/keras-team/keras-cv/blob/fc81ab52fca28ccfa5f1ec83523fbe2a4a177be3/keras_cv/models/stable_diffusion/stable_diffusion.py#L235-L238) from text. I think under normal circumstances, it is not part of the training process. While following [this](https://keras.io/examples/generative/finetune_stable_diffusion/) code example with your data (image-prompt), which components you think are absent for training or fine-tuning stable diffusion workflow?
> 
> ![2uC8rYJ](https://user-images.githubusercontent.com/81627860/238178576-6d843ba6-e994-421e-9753-90a6ed76c00a.png)

Thank you for your reply @innat-asj , I want to train the whole process from scratch, including CLIP, diffusion model, imageDecoder. But I don't quite understand how the latent encoding and decoding in the diffusion model are trained?

**Comment by qibin0506** [comment]
> > @qibin0506 cc. @jbischof It would be better to rephrase your main query. The decoding stage ([Decoder](https://github.com/keras-team/keras-cv/blob/fc81ab52fca28ccfa5f1ec83523fbe2a4a177be3/keras_cv/models/stable_diffusion/decoder.py#L28)) is only applicable in the inference steps, while [generating image](https://github.com/keras-team/keras-cv/blob/fc81ab52fca28ccfa5f1ec83523fbe2a4a177be3/keras_cv/models/stable_diffusion/stable_diffusion.py#L235-L238) from text. I think under normal circumstances, it is not part of the training process. While following [this](https://keras.io/examples/generative/finetune_stable_diffusion/) code example with your data (image-prompt), which components you think are absent for training or fine-tuning stable diffusion workflow?
> > ![2uC8rYJ](https://user-images.githubusercontent.com/81627860/238178576-6d843ba6-e994-421e-9753-90a6ed76c00a.png)
> 
> Thank you for your reply @innat-asj , I want to train the whole process from scratch, including CLIP, diffusion model, imageDecoder. But I don't quite understand how the latent encoding and decoding in the diffusion model are trained?

I found some useful information: https://keras.io/examples/generative/fine_tune_via_textual_inversion/

**Comment by ianstenbit** [comment]
@qibin0506 what is the use case here for training from scratch instead of fine-tuning?

Generally, training this sort of model is _extremely_ expensive and requires billions of training images.
Most people tend to fine-tune instead.

KerasCV doesn't currently have any workflow or tutorial for training this model from scratch as this is a very uncommon use. To my knowledge, nobody really trains it from scratch except for the creators of the model.

**Comment by qibin0506** [comment]
> @qibin0506 what is the use case here for training from scratch instead of fine-tuning?
> 
> Generally, training this sort of model is _extremely_ expensive and requires billions of training images. Most people tend to fine-tune instead.
> 
> KerasCV doesn't currently have any workflow or tutorial for training this model from scratch as this is a very uncommon use. To my knowledge, nobody really trains it from scratch except for the creators of the model.

At first I just wanted to understand the whole training process, now I understand that it is impossible to train such a large model from scratch.
